src/audio/aria/aria.c

```diff
@@ -264,16 +264,24 @@ static int aria_set_config(struct processing_module *mod, uint32_t param_id,
 			   size_t response_size)
 {
 	struct aria_data *cd = module_get_private_data(mod);
+	struct comp_dev *dev = mod->dev;
 
+	comp_info(dev, "aria_set_config()");
 	if (param_id == ARIA_SET_ATTENUATION) {
 		if (fragment_size != sizeof(uint32_t)) {
-			comp_err(mod->dev, "Illegal fragment_size = %d", fragment_size);
+			comp_err(dev, "Illegal fragment_size = %d", fragment_size);
 			return -EINVAL;
 		}
-		memcpy_s(&cd->att, sizeof(uint32_t), fragment, sizeof(uint32_t));
+		cd->att = (size_t)(*(uint32_t *)fragment);
```

Comment:
so presumably `fragment` is always aligned?

Reply:
I don't know it's ensured... should I then memcpy() it to a local uint32_t and then type cast it to Aria's component data. I think it was a mistake to use size_t for most of Aria's internal state. But I didn't want to change everything with this fix.

Reply:
It doesn't look like the align is ensured to be aligned. Even in handler.c ipc4_set_vendor_config_module_instance drv->ops.set_large_config() the type is char.  Maybe it's best that I change the Aria component data type to uint32_t and keep the memcpy().

Reply:
You could just fix the types instead ? Sounds like we need to do it anyway ?

Reply:
Yes, optimizing the code needs fixing the types.

Reply:
Fixed now, it needed only one aria.h edit to change the memcpy destination type.

Reply:
> It doesn't look like the align is ensured to be aligned. Even in handler.c ipc4_set_vendor_config_module_instance drv->ops.set_large_config() the type is char. Maybe it's best that I change the Aria component data type to uint32_t and keep the memcpy().

@singalsu there isn't a huge difference (supposedly on all 32-bit platforms, at least on Xtensa) between `size_t` and `uint32_t`, both are unsigned 32-bit integers. But if this should run on 64-bit ARM cores too, then there could be a difference

Reply:
It failed in testbench x86 build.